CI: Failed Travis jobs due to GBQ

Lots of failed jobs coming from Travis lately with tracebacks like this:

```python
    @pytest.fixture()
    def gbq_dataset(self):
        # Setup Dataset
        _skip_if_no_project_id()
        _skip_if_no_private_key_path()
    
        dataset_id = "pydata_pandas_bq_testing_" + generate_rand_str()
    
>       self.client = _get_client()

pandas/tests/io/test_gbq.py:160: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/tests/io/test_gbq.py:68: in _get_client
    return bigquery.Client(project=project_id, credentials=credentials)
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/google/cloud/bigquery/client.py:179: in __init__
    project=project, credentials=credentials, _http=_http
../../../miniconda3/envs/pandas-dev/lib/python3.7/site-packages/google/cloud/client.py:250: in __init__
    Client.__init__(self, credentials=credentials, client_options=client_options, _http=_http)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <google.cloud.bigquery.client.Client object at 0x7f5c46e157d0>
credentials = <google.oauth2.service_account.Credentials object at 0x7f5c46e15bd0>
_http = None
client_options = ClientOptions: {'api_endpoint': None, 'client_cert_source': None}

    def __init__(self, credentials=None, _http=None, client_options=None):
        if isinstance(client_options, dict):
            client_options = google.api_core.client_options.from_dict(client_options)
        if client_options is None:
            client_options = google.api_core.client_options.ClientOptions()
    
>       if credentials and client_options.credentials_file:
E       AttributeError: 'ClientOptions' object has no attribute 'credentials_file'
```

Line that's raising:

https://github.com/pandas-dev/pandas/blob/c8b44dda08ab3a2feb2c4572be960469c3f065f7/pandas/tests/io/test_gbq.py#L160

Sample Travis log:

https://api.travis-ci.org/v3/job/728073325/log.txt

cc @tswast if any ideas

--- a/ci/deps/travis-37-cov.yaml
+++ b/ci/deps/travis-37-cov.yaml
@@ -32,7 +32,7 @@ dependencies:
   - google-cloud-bigquery>=1.27.2 # GH 36436
   - psycopg2
   - pyarrow>=0.15.0
-  - pymysql
+  - pymysql=0.7.11
   - pytables
   - python-snappy
   - python-dateutil
@@ -40,7 +40,7 @@ dependencies:
   - s3fs>=0.4.0
   - scikit-learn
   - scipy
-  - sqlalchemy
+  - sqlalchemy=1.3.0
   - statsmodels
   - xarray
   - xlrd

--- a/ci/deps/travis-37-cov.yaml
+++ b/ci/deps/travis-37-cov.yaml
@@ -15,7 +15,6 @@ dependencies:
   # pandas dependencies
   - beautifulsoup4
   - botocore>=1.11
-  - cython>=0.29.16
   - dask
   - fastparquet>=0.3.2
   - fsspec>=0.7.4
@@ -31,12 +30,13 @@ dependencies:
   - odfpy
   - openpyxl
   - pandas-gbq
-  - google-cloud-bigquery>=1.27.2
+  - google-cloud-bigquery>=1.27.2 # GH 36436
   - psycopg2
   - pyarrow>=0.15.0
   - pymysql
   - pytables
   - python-snappy
+  - python-dateutil
   - pytz
   - s3fs>=0.4.0
   - scikit-learn
@@ -52,5 +52,4 @@ dependencies:
     - brotlipy
     - coverage
     - pandas-datareader
-    - python-dateutil
     - pyxlsb

--- a/ci/deps/travis-37-cov.yaml
+++ b/ci/deps/travis-37-cov.yaml
@@ -1,6 +1,5 @@
 name: pandas-dev
 channels:
-  - defaults
   - conda-forge
 dependencies:
   - python=3.7.*

--- a/ci/deps/travis-37-cov.yaml
+++ b/ci/deps/travis-37-cov.yaml
@@ -31,6 +31,7 @@ dependencies:
   - odfpy
   - openpyxl
   - pandas-gbq
+  - google-cloud-bigquery>=1.27.2
   - psycopg2
   - pyarrow>=0.15.0
   - pymysql

--- a/ci/deps/travis-37-locale.yaml
+++ b/ci/deps/travis-37-locale.yaml
@@ -29,7 +29,6 @@ dependencies:
   - google-cloud-bigquery>=1.27.2 # GH 36436
   - pyarrow>=0.17
   - psycopg2=2.7
-  - pyarrow>=0.15.0 # GH #35813
   - pymysql=0.7.11
   - pytables
   - python-dateutil

--- a/ci/deps/travis-37-locale.yaml
+++ b/ci/deps/travis-37-locale.yaml
@@ -26,7 +26,7 @@ dependencies:
   - numpy
   - openpyxl
   - pandas-gbq
-  - google-cloud-bigquery>=1.27.2
+  - google-cloud-bigquery>=1.27.2 # GH 36436
   - pyarrow>=0.17
   - psycopg2=2.7
   - pyarrow>=0.15.0 # GH #35813

--- a/ci/deps/travis-37-locale.yaml
+++ b/ci/deps/travis-37-locale.yaml
@@ -25,7 +25,8 @@ dependencies:
   - numexpr
   - numpy
   - openpyxl
-  - pandas-gbq=0.12.0
+  - pandas-gbq
+  - google-cloud-bigquery>=1.27.2
   - pyarrow>=0.17
   - psycopg2=2.7
   - pyarrow>=0.15.0 # GH #35813

